ledger-api/grpc-definitions/com/daml/ledger/api/v2/reassignment.proto

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) 2023 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+syntax = "proto3";
+
+package com.daml.ledger.api.v2;
+
+import "com/daml/ledger/api/v1/event.proto";
+import "google/protobuf/timestamp.proto";
+
+option java_outer_classname = "ReassignmentOuterClass";
+option java_package = "com.daml.ledger.api.v2";
+option csharp_namespace = "Com.Daml.Ledger.Api.V2";
+
+// Complete view of an on-ledger reassignment.
+message Reassignment {
+
+  // Assigned by the server. Useful for correlating logs.
+  // Must be a valid LedgerString (as described in ``value.proto``).
+  // Required
+  string update_id = 1;
+
+  // The ID of the command which resulted in this reassignment. Missing for everyone except the submitting party on the submitting participant.
+  // Must be a valid LedgerString (as described in ``value.proto``).
+  // Optional
+  string command_id = 2;
+
+  // The workflow ID used in reassignment command submission. Only set if the ``workflow_id`` for the command was set.
+  // Must be a valid LedgerString (as described in ``value.proto``).
+  // Optional
+  string workflow_id = 3;
+
+  // The absolute offset. The format of this field is described in ``ledger_offset.proto``.
+  // Required
+  string offset = 4;
+
+  oneof event {
+    UnassignedEvent unassigned_event = 5;
+    AssignedEvent assigned_event = 6;
+  }
+}
+
+// Records that a contract has been unassigned, and it becomes unusable on the source domain
+message UnassignedEvent {
+
+  // The ID of the unassignment. This needs to be used as an input for a assign ReassignmentCommand.
```

Comment:
should we document that (source domain, unassign_id) is unique? From a user perspective this is a pretty useful guarantee and we use it in our implementation. @rgugliel-da do you think that's reasonable as a public facing guarantee or do you expect to change that? Happy to keep unassign_id itself opaque, we don't need ordering or anything else on that, just uniqueness on the pair.

Reply:
Good point. I think we should mention it, especially because this is what is used in the assign command (so somehow you expect unicity)

Reply:
documented